frontends/common/resolveReferences/resolveReferences.cpp

```diff
@@ -180,6 +180,7 @@ ResolveReferences::ResolveReferences(ReferenceMap* refMap,
 void ResolveReferences::addToContext(const IR::INamespace* ns) {
     LOG1("Adding to context " << dbp(ns));
     BUG_CHECK(context != nullptr, "No resolution context; did not start at P4Program?");
+    checkShadowing(ns);
     context->push(ns);
```

Comment:
All I can tell you is that there are no new warnings on the public tests we have.
If this is too strict we can refine it later.